src/common/utils.cpp

```diff
@@ -818,6 +818,43 @@ std::string trim(std::string const& str, std::string const& whitespace)
     return str.substr(strBegin, strRange);
 }
 
+/// Matches the given target with the given pattern, using the algorithm linked below.
+/// There's possibly a more clever / modern c way of doing this.
+/// https://www.geeksforgeeks.org/wildcard-character-matching/
+bool matchesRecursive(const char* pattern, const char* target, const char* wildcard) {
+    // If we reach at the end of both strings, we are done
+    if (*pattern == '\0' && *target == '\0')
```

Comment:
What if either of these are passed in as nullptr?

Reply:
https://www.youtube.com/watch?v=bLHL75H_VEM

Reply:
Good catch. I didn't due my due diligence and mostly copied over the linked algorithm. (shame on me)

Reply:
Since this method is now a lambda inside matches, and matches uses std::string to convert em to const char*, this should no longer be possible, but I am going to add the check to be safe.

Reply:
Done